Add documentation for Day 2-7 Activation, Google Play Store and Apple App Store (#545)
mozilla/firefox-data-docs: src/datasets/non_desktop/google_play_store/reference.md

# Google Play Store
<!-- toc -->

# Introduction

> The [`google_play_store`](https://console.cloud.google.com/bigquery?project=moz-fx-data-marketing-prod&folder=&organizationId=&p=moz-fx-data-marketing-prod&d=google_play_store&page=dataset) dataset is used to understand the acquisition performance for the nondesktop products through the Google Play Store along the key metrics and dimensions. Google’s documentation for all metrics and dimensions can be found [here](https://support.google.com/googleplay/android-developer/answer/6263332?hl=en).

Reviewer: ```suggestion
The [`google_play_store`](https://console.cloud.google.com/bigquery?project=moz-fx-data-marketing-prod&folder=&organizationId=&p=moz-fx-data-marketing-prod&d=google_play_store&page=dataset) dataset is used to understand the acquisition performance for mobile products on the Google Play Store along key metrics and dimensions. Google’s documentation for all metrics and dimensions can be found [here](on the Play Store support portal).
```

# Contents
The [`google_play_store`](https://console.cloud.google.com/bigquery?project=moz-fx-data-marketing-prod&folder=&organizationId=&p=moz-fx-data-marketing-prod&d=google_play_store&page=dataset) dataset contains a collection of tables and views exported from Google. The key tables currently being used for nondesktop reporting are:

> - [`p_Retained_installers_channel_v1`](https://console.cloud.google.com/bigquery?project=moz-fx-data-marketing-prod&p=moz-fx-data-marketing-prod&d=google_play_store&t=p_Retained_installers_channel_v1&page=table) - Store activity by acquisition channels. Acquisition channels include:

Author: changed in new commit

    - **Google Search (organic)** - Unique users who visited the app’s store listing on the Play Store app from a Google Search
    - **Google Ads** - Unique users who visited the app’s store listing on the Play Store app from a Google Ads ad. Data between Google Ads and the Play Console can differ. 
    - **Other**
- [`p_Retained_installers_country_v1`](https://console.cloud.google.com/bigquery?project=moz-fx-data-marketing-prod&p=moz-fx-data-marketing-prod&d=google_play_store&t=p_Retained_installers_country_v1&page=table) - Store activity by country.
- [`p_Retained_installers_play_country_v1`](https://console.cloud.google.com/bigquery?project=moz-fx-data-marketing-prod&p=moz-fx-data-marketing-prod&d=google_play_store&t=p_Retained_installers_play_country_v1&page=table) - Store activity for the Play  Store (organic) acquisition channel. Includes a country breakdown for the channel.
- [`p_Retained_installers_utm_tagged_v1`](https://console.cloud.google.com/bigquery?project=moz-fx-data-marketing-prod&p=moz-fx-data-marketing-prod&d=google_play_store&t=p_Retained_installers_utm_tagged_v1&page=table) - Store activity for the Tracked Channels (UTM) acquisition channel by campaign UTM.

> There are other tables available that are yet to be explored / incorporated into regular reporting. Some that may be of interest include: (Crashes table by app_version, device and os; installs tables and ratings tables).

Reviewer: ```suggestion
There are other tables available that are yet to be explored / incorporated into regular reporting. Some that may be of interest include:

- Crashes by app_version
- Device and OS
- Installs
- Ratings
```

- **Installers_retained_for_30_days** - Installers who kept the app on at least one of their devices for 30 days. Installation doesn’t mean the app was opened over this period. 
- **Installers_to_30_days_retention_rate** - Percentage of installers who have the app on one of their devices 30 days after install.

## Background and Caveats

As of Aug 25 2020 not all the tables available in the dataset have been explored and vetted for accuracy by the data science team. Tables that were fully reviewed and being documented here are the Retained_installers tables whose primary use case is to explain acquisition. 

**Note:** Google does not make the play store data available for export every day. The export job checks for new files every day. However, having monitored the job, it appears the data is made available every 7 - 14 days, and seems to primarily be made available on weekends. Due to this lack of consistency, there will be delays in the data available for this dataset. The data currently in BigQuery is the most current data available from Google.

## Accessing the Data

Access the data at moz-fx-data-marketing-prod.google_play_store

# Data Reference

> ## Schema

Reviewer: Does this schema cover all tables or just one of them? If the latter, maybe we can just leave this section out. This tends to go out of date super quickly anyway...

Author: Covers all the tables. The dimension is the only thing that changes but metrics stay the same for the retained_installer tables.

```sql
SELECT
  Date,
  Package_Name,
  Country,
  SUM(Store_Listing_Visitors) as Store_Visits,
  SUM(Installers) as installs,
  SAFE_DIVIDE(SUM(Installers), SUM(Store_Listing_Visitors)) as install_rate
FROM
  `moz-fx-data-marketing-prod.google_play_store.p_Retained_installers_country_v1`
WHERE
  Date = "2020-08-01"
  AND Package_Name = "org.mozilla.firefox"
GROUP BY
  date, Package_name, Country
ORDER BY
  Store_Visits DESC
  ```

##### Calculate App Store Activity for a given day by source and app 

```sql
SELECT
  Date,
  Package_Name,
  Acquisition_Channel,
  SUM(Store_Listing_Visitors) as Store_Visits,
  SUM(Installers) as installs,
  SAFE_DIVIDE(SUM(Installers), SUM(Store_Listing_Visitors)) as install_rate
FROM
  `moz-fx-data-marketing-prod.google_play_store.p_Retained_installers_channel_v1`
WHERE
  Date = "2020-08-01"
GROUP BY
  date, Package_name, Acquisition_Channel
ORDER BY
  package_name, Store_Visits
```

## Scheduling
The job to retrieve the raw data from the Google Play Store can be found in [github](https://github.com/mozilla/play-store-export) and it is scheduled in [airflow](https://github.com/mozilla/telemetry-airflow/blob/master/dags/play_store_export.py).

> **NOTE:** Google does not make the play store data available for export every day. The export job checks for new files every day. However, having monitored the job, it appears the data is made available every 7 - 14 days, and seems to primarily be made available on the weekends. Due to this lack of consistency, there will be delays in the data available for this dataset. The data currently in BigQuery is the most current data available from Google.

Author: You're right. Deleted it.

> 

Reviewer: ```suggestion
```
